shared/feature/feedings/src/main/java/com/epmedu/animeal/feedings/presentation/FeedingsScreenUi.kt

```diff
@@ -48,12 +63,55 @@ internal fun FeedingsScreenUI(
             }
 
             else -> {
-                FeedingList(state.feedings)
+                FeedingsCategoryTab(state.feedingsFiltered, state.feedingsCategory, onFilterClick)
             }
         }
     }
 }
 
+@Composable
+fun FeedingsCategoryTab(
+    feedings: ImmutableList<FeedingModel>,
+    feedingsCategory: FeedingFilterCategory,
+    onFilterClick: (FeedingFilterCategory) -> Unit
+) {
+    var selectedTab by remember { mutableStateOf(feedingsCategory) }
```

Comment:
Is changing tabs faster this way?

Reply:
Yes, also this way keeps ui changes and feedings list population process independent.